torch/nn/modules/loss.py

```diff
@@ -1246,6 +1251,125 @@ def forward(self, anchor: Tensor, positive: Tensor, negative: Tensor) -> Tensor:
                                      eps=self.eps, swap=self.swap, reduction=self.reduction)
 
 
+class TripletMarginWithDistanceLoss(_Loss):
+    r"""Creates a criterion that measures the triplet loss given input
+    tensors :math:`a`, :math:`p`, and :math:`n` (representing anchor,
+    positive, and negative examples, respectively); and a nonnegative,
+    real-valued function ("distance function") used to compute the relationship
+    between the anchor and positive examples ("positive distance") and the
+    anchor and negative examples ("negative distance").
+
+    The unreduced loss (i.e., with `reduction` set to `'none'`)
+    can be described as:
+
+    .. math::
+        \ell(a, p, n) = L = \{l_1,\dots,l_N\}^\top, \quad
+        l_i = \max \{d(a_i, p_i) - d(a_i, n_i) + {\rm margin}, 0\}
+
+    where :math:`N` is the batch size; :math:`d` is a nonnegative, real-valued function
+    quantifying the relationship between two tensors, referred to as `distance_function`;
+    and :math:`margin` is a non-negative margin between the positive and negative
+    distances that is required for a 0 loss.  The input tensors have :math:`N` elements
+    each and can be of any shape that the distance function can handle.
+
+    If :attr:`reduction` is not ``'none'``
+    (default ``'mean'``), then:
+
+    .. math::
+        \ell(x, y) =
+        \begin{cases}
+            \operatorname{mean}(L), &  \text{if reduction} = \text{`mean';}\\
+            \operatorname{sum}(L),  &  \text{if reduction} = \text{`sum'.}
+        \end{cases}
+
+    See also :class:`~torch.nn.TripletMarginLoss`, which computes the triplet
+    loss for input tensors using the :math:`l_p` distance as the distance function.
+
+    Note: does not support JIT scripting.
+
+    Args:
+        distance_function (callable, optional): A nonnegative, real-valued function that
+            quantifies the relationship between two tensors. If not specified,
+            `nn.PairwiseDistance` will be used.  Default: ``None``
+        is_similarity_function (bool, optional): Whether `distance_function` represents a
+            similarity metric, i.e., larger values are closer. If True, computes the difference of
```

Comment:
This would me clearer to me if it was reading:
"""
Whether `distance_function` represents a similarity metric, i.e., larger values are closer. Otherwise it is a distance metric, i.e., smaller values are closer.
"""
Do we need to specify more how we use it?

Reply:
Will expand that.  I think the crux of the difference -- reversing the difference of distances -- is already there, and the standard equation is included in the main doc body